onnxruntime/core/graph/function_impl.h

```diff
@@ -34,7 +34,7 @@ class FunctionImpl final : public Function {
  private:
   const onnxruntime::Graph* const parent_graph_;
   std::unique_ptr<ONNX_NAMESPACE::OpSchema> op_schema_;
-  onnxruntime::Model body_;
+  std::unique_ptr<onnxruntime::Model> body_;
```

Comment:
> std::unique_ptr<onnxruntime::Model> body_; [](start = 1, length = 43)

It looks like this ptr does not really represent an optional member. However, it does affect the ability of the class to be copied since unique_ptr is not copyable. It may not matter now but may matter later.
So I would simply construct a new Model on the stack in the constructor and move it into the default constructed member.

Reply:
Per offline discussion resolving this without any changes. (Model does not expose its default constructor + model constructor calls graph resolve therefore in order to successfully create a model for function body it is necessary to first construct the opset import map.)